hack/update-docs.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Tekton Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source $(dirname $0)/../vendor/github.com/tektoncd/plumbing/scripts/library.sh
+cd ${REPO_ROOT_DIR}
+
+# Whether this script is being used for validation or not
+DIFF=
+# Syntax format for code fences
+SYNTAX="YAML"
+# All code fences encountered will be ignored
+FENCE_ON='^```'${SYNTAX}'$'
+FENCE_OFF='^```$'
+# Comment that will specify the file to embed within $SYNTAX code fence
+COMMENT='<!-- *FILE: *(.*) *-->'
+
+# Parse flags
+while getopts ":d" opt; do
+  case ${opt} in
+    d )
+        DIFF="true"
+        ;;
+    * )
+        echo "Invalid Option: -$OPTARG" 1>&2
+        exit 1
+        ;;
+  esac
+done
+shift $((OPTIND -1))
+
+doc_files="$(find docs -name "*.md")"
+for file in ${doc_files};do
+    new_file="${file%/*}/updated-${file##*/}"
+    > ${new_file}
+    fenced="false"
+    # Read each markdown file for replacements
+    while IFS= read -r line; do
+        if [[ ${line} =~ ${FENCE_ON} ]];then
+            fenced="true"
+            continue
+        fi
+        if [[ ${line} =~ ${FENCE_OFF} ]];then
+            fenced="false"
+            continue
+        fi
+        # Write to replacement file
+        if [[ ${fenced} == false ]];then 
+            # Copy line
+            echo "${line}" >> ${new_file}
+            # Inline file
+            if [[ "$line" =~ ${COMMENT} ]];then
+                echo '```'$SYNTAX >> ${new_file}
+                cat ${BASH_REMATCH[1]} >> ${new_file}
```

Comment:
nit: Should this report an error if the file can't be found?

Reply:
The `cat` utility should  output if any file does not exist like this:
```
cat fakefile1.md >> fakefile2.md
cat: fakefile1.md: No such file or directory
```

Reply:
Small nit here is that if the script errors, you'll be left with those `update-*.md` files. One change could be to use this cleanup function and trap on EXIT:
```
function cleanup {
  tmp_files=$(find docs -name "${TMP_PREFIX}*.md")
  for file in ${tmp_files}; do
    rm "${file}"
  done
}

trap cleanup EXIT
```

If you make this change, then the temp prefix should probably be something more unique than `update`; for example:
```
TMP_PREFIX='pleasedontusethistmpprefix'
```